packages/flutter/lib/src/rendering/slivers.dart

```diff
@@ -0,0 +1,1850 @@
+             '$userScrollDirection, '
+             'scrollOffset: ${scrollOffset.toStringAsFixed(1)}, '
+             'remainingPaintExtent: ${remainingPaintExtent.toStringAsFixed(1)}, ' +
+             (overlap != 0.0 ? 'overlap: ${overlap.toStringAsFixed(1)}, ' : '') +
+             'crossAxisExtent: ${crossAxisExtent.toStringAsFixed(1)}'
+           ')';
+  }
+}
+
+class SliverGeometry {
+  SliverGeometry({
+    this.scrollExtent: 0.0,
+    this.paintExtent: 0.0,
+    double layoutExtent,
+    double opaqueExtent,
+    this.maxPaintExtent: 0.0,
+    double hitTestExtent,
+    bool visible,
+    this.scrollOffsetCorrection: 0.0
+  }) : opaqueExtent = opaqueExtent ?? paintExtent,
+       layoutExtent = layoutExtent ?? paintExtent,
+       hitTestExtent = hitTestExtent ?? paintExtent,
+       visible = visible ?? paintExtent > 0.0 {
+    assert(debugAssertIsValid);
+  }
+
+  static final SliverGeometry zero = new SliverGeometry();
+
+  /// The (estimated) total scroll extent that this sliver has content for. In
+  /// other words, the scroll offset of the end of the last bit of content of
+  /// this sliver.
+  ///
+  /// This value must be accurate if the [paintExtent] is less than the
+  /// [SliverConstraints.remainingPaintExtent] provided during layout.
+  final double scrollExtent;
+
+  /// The amount of visual space that was taken by the sliver to render the
+  /// subset of the sliver that covers all or part of the
+  /// [SliverConstraints.remainingPaintExtent].
+  ///
+  /// This must be between zero and [SliverConstraints.remainingPaintExtent].
+  ///
+  /// This contributes to the calculation for the next sliver's
+  /// [SliverConstraints.overlap].
+  final double paintExtent;
+
+  /// The distance from the first visible part of this sliver to the first
+  /// visible part of the next sliver, assuming the next sliver's
+  /// [SliverConstraints.scrollOffset] is zero.
+  ///
+  /// This must be between zero and [paintExtent]. It defaults to [paintExtent].
+  final double layoutExtent;
+
+  /// The amount of visual space that was fully painted by this sliver, and does
+  /// not need to be also painted by the next. If this is less than the
+  /// [layoutExtent], then the start scroll offset of the next sliver (if it
+  /// would not be zero) will be offset by the difference, so that it paints
+  /// under this sliver as well.
+  ///
+  /// This must be between zero and [paintExtent]. It defaults to [paintExtent].
+  /// Typically it is [paintExtent] for slivers that scroll synchronously with
+  /// the rest of the viewport's contents, and zero for slivers that are pinned,
+  /// floating, or otherwise act in a non-synchronous fashion.
+  ///
+  /// For slivers that act in a synchronous fashion it is usually effectively
+  /// ignored since those will either not paint at all (if they are scrolled
+  /// past the leading edge), or the next sliver will be at scroll offset zero.
+  final double opaqueExtent;
```

Comment:
Never mind. I'm removing opaqueExtent entirely. Careful examination revealed the embarrassing truth that it was never actually used.